### PR TITLE
Remake bzip2 using xmake.lua

### PR DESCRIPTION
**Pull request by SirLynix**
bzip2 didn't support dynamic build and didn't compile on mingw, so I remade it using a xmake.lua that should compile on every platform

### CODE REVIEW

**Comment by SirLynix** [comment]
@waruqi I removed this line
```lua
        package:addenv("PATH", "bin")
```

because I assumed xmake does it automatically, but I'm not sure about it, should I add it back in on_load?

**Comment by waruqi** [comment]
> @waruqi I removed this line
> 
> ```lua
>         package:addenv("PATH", "bin")
> ```
> 
> because I assumed xmake does it automatically, but I'm not sure about it, should I add it back in on_load/on_install?

You need add it. only binary package (`set_kind("binary") in package()`) will add it automatically.

File: packages/b/bzip2/xmake.lua [hunk 1]
```diff
@@ -14,6 +14,8 @@ package("bzip2")
     end)
 
     on_install(function (package)
+        package:addenv("PATH", "bin")
```
Comment (waruqi) [inline packages/b/bzip2/xmake.lua:17]: you can move to on_load

File: packages/b/bzip2/port/xmake.lua [hunk 2]
```diff
@@ -18,38 +22,42 @@ target("bz2")
...
-            os.ln(path.join(binarydir, "bzmore"), path.join(binarydir, "bzless"))
-        end
-    end)
-
-target("bzip2recover")
-    set_kind("binary")
-    add_deps("bz2")
-    add_files("bzip2recover.c")
+if has_config("build_bin") then
+
```
Comment (waruqi) [inline packages/b/bzip2/port/xmake.lua:26]: we can also use `set_default(false)` to disable it by default. then we use `xmake build --all` to build all

Comment (waruqi) [reply packages/b/bzip2/port/xmake.lua:26]: oh, but package.tools.xmake has not support it yet. = =

**Comment by xq114** [comment]
The patch is not needed, and it has caused problems with other packages like libarchive. bzip2 already has a def file libbz2.def, `__declspec(dllexport)` is not required actually. I'm reverting the patch now (https://github.com/xmake-io/xmake-repo/pull/702), if it does affect your code, please make me aware.

**Pull request merged**